signer/src/storage/model.rs

```diff
@@ -1352,6 +1349,260 @@ pub struct WithdrawalRejectEvent {
     pub signer_bitmap: BitArray<[u8; 16]>,
 }
 
+impl From<u8> for BitcoinBlockHeight {
+    fn from(value: u8) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u16> for BitcoinBlockHeight {
+    fn from(value: u16) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u32> for BitcoinBlockHeight {
+    fn from(value: u32) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u64> for BitcoinBlockHeight {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+impl From<usize> for BitcoinBlockHeight {
+    fn from(value: usize) -> Self {
+        Self(value as u64)
+    }
+}
+
+// Conversion u64 => StacksBlockHeight is not implemented intentionally.
+// Use deref instead.
+
+impl From<BitcoinBlockHeight> for u128 {
+    fn from(value: BitcoinBlockHeight) -> Self {
+        *value as u128
+    }
+}
+
+impl std::fmt::Display for BitcoinBlockHeight {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+impl Deref for BitcoinBlockHeight {
+    type Target = u64;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+impl TryFrom<BitcoinBlockHeight> for i64 {
+    type Error = TryFromIntError;
+    fn try_from(value: BitcoinBlockHeight) -> Result<Self, Self::Error> {
+        i64::try_from(value.0)
+    }
+}
+
+impl TryFrom<i64> for BitcoinBlockHeight {
+    type Error = TryFromIntError;
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        Ok(Self(<u64>::try_from(value)?))
```

Comment:
nit? also for stacks one
```suggestion
        u64::try_from(value).map(Self)
```

Reply:
WoW, nice!

Reply:
[4f2167b](https://github.com/stacks-sbtc/sbtc/pull/1448/commits/4f2167bfe8773652452d5b37a9e27fe406b9e964)